librepo/handle.c

```diff
@@ -841,7 +867,57 @@ lr_handle_setopt(LrHandle *handle,
     return ret;
 }
 
-static gboolean
+gboolean
+lr_handle_network_wait(LrHandle *handle, GError **err, guint seconds)
+{
+
+    assert(!err || *err == NULL);
+
+    if (!handle) {
+        g_set_error(err, LR_HANDLE_ERROR, LRE_BADFUNCARG,
+                    "No handle specified");
+        return FALSE;
+    }
+
+    GNetworkMonitor *monitor = g_network_monitor_get_default();
+    GCancellable *cancellable = g_cancellable_new ();
+
+    struct callback_data data_struct;
+    data_struct.input_seconds = seconds;
+    data_struct.begin = time(NULL);
+    data_struct.cancellable = cancellable;
+    data_struct.monitor = monitor;
+
+    gchar *baseurl;
+    if(handle->metalinkurl)
+      baseurl = handle->metalinkurl;
+    else if(handle->mirrorlisturl)
+      baseurl = handle->mirrorlisturl;
+    else if(handle->urls)
+      baseurl = handle->urls[0];
+    assert(baseurl);
+
+    GUri *uri = g_uri_parse(baseurl, G_URI_FLAGS_NONE, NULL);
```

Comment:
I think one thing we should to do here is call `g_uri_get_scheme()` and if it's `file` then we bypass all of this.